src/transforms/onehotencoding.jl

```diff
@@ -0,0 +1,86 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    OneHotEncoding(col)
+    
+Transforms categorical column `col` into one-hot columns of levels
+returned by the `levels` function of CategoricalArrays.jl.
+
+# Examples
+
+```julia
+OneHotEncoding(1)
+OneHotEncoding(:a)
+OneHotEncoding("a")
+```
+"""
+struct OneHotEncoding{S<:ColSelector} <: Stateless
+  col::S
+end
+
+isrevertible(::Type{<:OneHotEncoding}) = true
+
+_colname(col::Integer, names) = names[col]
+_colname(col::AbstractString, names) =
+  _colname(Symbol(col), names)
+
+function _colname(col::Symbol, names)
+  @assert col ∈ names "Invalid column."
+  return col
+end
+
+_name(nm, names) = 
+  nm ∈ names ? _name(Symbol("$(nm)_"), names) : nm
+
+function apply(transform::OneHotEncoding, table)
+  cols = Tables.columns(table)
+  names = collect(Tables.columnnames(cols))
+  columns = AbstractVector[Tables.getcolumn(cols, nm) for nm in names]
```

Comment:
Is this related to the issue we had in other transform where we are avoiding specialization of type in order to revert later? Why do we need the AbstractVector here? Maybe we should use Any as we did in the other case?

Reply:
So I think Any might be a better choice.